### ISSUE TITLE
Auto Generation of Vision and Light Blocking Polylines [FEAT]

### ISSUE THREAD

**Issue by chisaipete** [Kruptein/PlanarAlly#327]
**Is your feature request related to a problem? Please describe.**
No problem.

**Describe the solution you'd like**
Just like how the map tool asks you to select an area and then resizes an asset on the map layer, what if there was a way to select a color in a map and have it draw polygon lines following the color in the map.  This would work great for maps with big bold lines for edges. The user then could select and delete the lines they don't want on the map.  This would speed up drawing vision/light blocking significantly.

**Describe alternatives you've considered**
Drawing the shapes by hand is certainly already pretty quick, so this is more of just a "hmm, could we do that?" sort of question.

**Additional context**
This probably wouldn't apply for every kind of imported map.  Some full color maps MAY be harder to auto-detect boundary lines/shapes.

**Comment by ekerstens** [comment]
Would this be related to https://github.com/Kruptein/PlanarAlly/issues/641? Sounds similar to adding the functionality into the tool to generate the svg layer based on a monochrome map asset.

**Comment by PalasX** [comment]
oh man, how did i ever miss this feature request! Looking back, doing it the way described in this request sounds like implementing edge detection, raster to vector converstion, and vector to shape conversion all in javascript. feel sorry for the dev that ever has to tackle that ;P hehe.

For reference, this works really well, at least for donjon maps:
https://github.com/Kruptein/PlanarAlly/issues/641#issuecomment-817076144

**Comment by Kruptein** [comment]
Yeah for now I think it's going to be more interesting to use an external svg generator for now.

It would be very cool to have this built-in but it's indeed quite a challenge, worth coming back to in the future
